### DISCUSSION TITLE
What am I doing when I press Ctrl+Z?

### DISCUSSION THREAD

**Discussion by bie-zheng** [mortii/anki-morphs#293] [Q&A] [answered]
When I press Ctrl+Z, regardless of whether I am in the selected deck of Ankimorphs or not, the revoked "AnkiMorphs custom undo" will pop up.I can only go back to the previous card when I press Ctrl+Z for the second time.
![图片](https://github.com/user-attachments/assets/6be72e00-b32a-4bbe-a391-e60acec242a4)
So I want to know what this means, thanks very much if anyone can answer it.windows11 24.06 qt6.

**Comment by mortii** [comment, marked as answer]
When you answer a card, any morphs on that card will be stored in a 'morphs seen today' list. When undoing (ctrl+z), that list becomes outdated, which can cause unexpected skipping of future cards. To avoid this, we've added custom logic to the undo process, which rebuilds the 'morphs seen today' list.

>  I can only go back to the previous card when I press Ctrl+Z for the second time.

It's a pretty complicated process so I wouldn't be shocked if some bugs have been introduced because of it... Could you try to disable the add-on and repeat the same process to see if the result is any different?

**Reply by bie-zheng** [reply]
Thank you very much for your answer. I haven't found any other bugs yet. This is a great plugin!

**Reply by mortii** [reply]
@bie-zheng I'm not able to reproduce your problem of having to press ctrl+z multiple times to get back to the previous card. When I do it, it says `answer card undone` immediately. If you are able to consistently reproduce the bug, please create a bug issue with the steps you used. Thanks!
